src/ipc-types.d.ts

```diff
@@ -11,6 +11,12 @@ interface ShowNotificationOptions extends Electron.NotificationConstructorOption
 	showIcon: boolean;
 }
 
+enum ImportState {
+	Initial = 0,
+	Importing = 1,
+	Imported = 2,
+}
```

Comment:
Could we use the type exported in `lib/import-export/import/types.ts`?

Reply:
That's something I tried first, but importing or exporting anything from/to `src/ipc-types.d.ts` results in a type error for the rest of types used in this file, I guess due to a cyclic dependency, and `npm run start` does not work anymore. Do you experience the same?
All types on `src/ipc-types.d.ts` are shared across the repository, but in this case I'm also using ImportState as a value.

Do you know any other way to achieve that? For now I've decided to remove the duplicated Enum and use the literals instead:
https://github.com/Automattic/studio/pull/369/commits/3a64bcca477f3bbadde88a8dc2ec9565ae610426

Reply:
Yep, there are some caveats when using Enums ([reference](https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls)). In this case, I think we can solve it by using `declare` when defining the enum in `src/ipc-types.d.ts`. E.g.:

```patch
diff --git forkSrcPrefix/src/ipc-types.d.ts forkDstPrefix/src/ipc-types.d.ts
index 9b12bbc01a2689ab004b9a63d28c8998ea7e9469..b639575d51b7b95134477c6c8ac7cff38a9b2b5a 100644
--- forkSrcPrefix/src/ipc-types.d.ts
+++ forkDstPrefix/src/ipc-types.d.ts
@@ -11,6 +11,12 @@ interface ShowNotificationOptions extends Electron.NotificationConstructorOption
 	showIcon: boolean;
 }
 
+declare enum ImportState {
+	Initial = 0,
+	Importing = 1,
+	Imported = 2,
+}
+
 interface StoppedSiteDetails {
 	running: false;
 
@@ -29,7 +35,7 @@ interface StoppedSiteDetails {
 		supportsMenus: boolean;
 	};
 	isAddingSite?: boolean;
-	importState?: 0 | 1 | 2;
+	importState?: ImportState;
 }
 
 interface StartedSiteDetails extends StoppedSiteDetails {
diff --git forkSrcPrefix/src/lib/import-export/import/types.ts forkDstPrefix/src/lib/import-export/import/types.ts
index cc8b0b961a9e4ac75cb467ff958db1d3e21c2474..8687feba85f76f8a2ccc4dff28502c848bc8fd34 100644
--- forkSrcPrefix/src/lib/import-export/import/types.ts
+++ forkDstPrefix/src/lib/import-export/import/types.ts
@@ -23,10 +23,4 @@ export interface BackupArchiveInfo {
 	type: string;
 }
 
-export enum ImportState {
-	Initial = 0,
-	Importing = 1,
-	Imported = 2,
-}
-
 export type NewImporter = new ( backup: BackupContents ) => Importer;
diff --git forkSrcPrefix/src/components/content-tab-import-export.tsx forkDstPrefix/src/components/content-tab-import-export.tsx
index a97f0beb3a0700b7166977abd9bd3aa91629c25c..594dfe3bc686ba5f59c6bf41b2ebb7f38e07ac2e 100644
--- forkSrcPrefix/src/components/content-tab-import-export.tsx
+++ forkDstPrefix/src/components/content-tab-import-export.tsx
@@ -14,7 +14,6 @@ import { cx } from '../lib/cx';
 import { sanitizeFolderName } from '../lib/generate-site-name';
 import { getIpcApi } from '../lib/get-ipc-api';
 import { ExportOptions } from '../lib/import-export/export/types';
-import { ImportState } from '../lib/import-export/import/types';
 import Button from './button';
 import { ProgressBarWithAutoIncrement } from './progress-bar';
 
diff --git forkSrcPrefix/src/components/tests/content-tab-import-export.test.tsx forkDstPrefix/src/components/tests/content-tab-import-export.test.tsx
index d69ca6a1f30911732e18dbbe8cffae7a0a498f21..2988ef496c3ab93f8e8bb9007501cc5a03cc4b61 100644
--- forkSrcPrefix/src/components/tests/content-tab-import-export.test.tsx
+++ forkDstPrefix/src/components/tests/content-tab-import-export.test.tsx
@@ -2,7 +2,6 @@ import { render, fireEvent, waitFor, screen, createEvent, act } from '@testing-l
 import { userEvent } from '@testing-library/user-event';
 import { useSiteDetails } from '../../hooks/use-site-details';
 import { getIpcApi } from '../../lib/get-ipc-api';
-import { ImportState } from '../../lib/import-export/import/types';
 import { ContentTabImportExport } from '../content-tab-import-export';
 
 jest.mock( '../../lib/get-ipc-api' );
diff --git forkSrcPrefix/src/hooks/use-site-details.tsx forkDstPrefix/src/hooks/use-site-details.tsx
index 3a908384933900fce168840cbcb66d1a77aa5b5a..16062aa67d8d1eeb9b5a4a7938c02f7a541387aa 100644
--- forkSrcPrefix/src/hooks/use-site-details.tsx
+++ forkDstPrefix/src/hooks/use-site-details.tsx
@@ -10,7 +10,7 @@ import {
 	useState,
 } from 'react';
 import { getIpcApi } from '../lib/get-ipc-api';
-import { BackupArchiveInfo, ImportState } from '../lib/import-export/import/types';
+import { BackupArchiveInfo } from '../lib/import-export/import/types';
 import { sortSites } from '../lib/sort-sites';
 import { useSnapshots } from './use-snapshots';
 
```

Let me know if that works 🤞 .

Reply:
Although TypeScript doesn't complain, I noticed that this approach doesn't work in runtime 😞.

Reply:
I haven't found a better way than your original approach by duplicating. However, an alternative could be to stop using an enum and have string values defined in a type. Here's a potential approach:

```patch
diff --git forkSrcPrefix/src/components/tests/content-tab-import-export.test.tsx forkDstPrefix/src/components/tests/content-tab-import-export.test.tsx
index d69ca6a1f30911732e18dbbe8cffae7a0a498f21..7191147823793f1345ebb193559a6af8260151a8 100644
--- forkSrcPrefix/src/components/tests/content-tab-import-export.test.tsx
+++ forkDstPrefix/src/components/tests/content-tab-import-export.test.tsx
@@ -2,7 +2,6 @@ import { render, fireEvent, waitFor, screen, createEvent, act } from '@testing-l
 import { userEvent } from '@testing-library/user-event';
 import { useSiteDetails } from '../../hooks/use-site-details';
 import { getIpcApi } from '../../lib/get-ipc-api';
-import { ImportState } from '../../lib/import-export/import/types';
 import { ContentTabImportExport } from '../content-tab-import-export';
 
 jest.mock( '../../lib/get-ipc-api' );
@@ -15,7 +14,7 @@ const selectedSite: SiteDetails = {
 	path: '/test-site',
 	phpVersion: '8.0',
 	adminPassword: btoa( 'test-password' ),
-	importState: ImportState.Initial,
+	importState: 'initial',
 };
 
 describe( 'ContentTabImportExport Import', () => {
diff --git forkSrcPrefix/src/components/content-tab-import-export.tsx forkDstPrefix/src/components/content-tab-import-export.tsx
index a97f0beb3a0700b7166977abd9bd3aa91629c25c..1057db67b46eaf7817d84dd68970e737435fc19c 100644
--- forkSrcPrefix/src/components/content-tab-import-export.tsx
+++ forkDstPrefix/src/components/content-tab-import-export.tsx
@@ -14,7 +14,6 @@ import { cx } from '../lib/cx';
 import { sanitizeFolderName } from '../lib/generate-site-name';
 import { getIpcApi } from '../lib/get-ipc-api';
 import { ExportOptions } from '../lib/import-export/export/types';
-import { ImportState } from '../lib/import-export/import/types';
 import Button from './button';
 import { ProgressBarWithAutoIncrement } from './progress-bar';
 
@@ -174,15 +173,15 @@ const ImportSite = ( props: { selectedSite: SiteDetails } ) => {
 		}
 	};
 	const clearImportState = () => {
-		updateSite( { ...props.selectedSite, importState: ImportState.Initial } );
+		updateSite( { ...props.selectedSite, importState: 'initial' } );
 		clearImportFileInput();
 	};
 
 	const startLoadingCursorClassName =
 		loadingServer[ props.selectedSite.id ] && 'animate-pulse duration-100 cursor-wait';
 
-	const isImporting = props.selectedSite.importState === ImportState.Importing && ! isDraggingOver;
-	const isImported = props.selectedSite.importState === ImportState.Imported && ! isDraggingOver;
+	const isImporting = props.selectedSite.importState === 'importing' && ! isDraggingOver;
+	const isImported = props.selectedSite.importState === 'imported' && ! isDraggingOver;
 	const isInitial = ! isImporting && ! isImported;
 	return (
 		<div className={ cx( 'flex flex-col w-full', startLoadingCursorClassName ) }>
diff --git forkSrcPrefix/src/lib/import-export/import/types.ts forkDstPrefix/src/lib/import-export/import/types.ts
index cc8b0b961a9e4ac75cb467ff958db1d3e21c2474..8687feba85f76f8a2ccc4dff28502c848bc8fd34 100644
--- forkSrcPrefix/src/lib/import-export/import/types.ts
+++ forkDstPrefix/src/lib/import-export/import/types.ts
@@ -23,10 +23,4 @@ export interface BackupArchiveInfo {
 	type: string;
 }
 
-export enum ImportState {
-	Initial = 0,
-	Importing = 1,
-	Imported = 2,
-}
-
 export type NewImporter = new ( backup: BackupContents ) => Importer;
diff --git forkSrcPrefix/src/ipc-types.d.ts forkDstPrefix/src/ipc-types.d.ts
index 9b12bbc01a2689ab004b9a63d28c8998ea7e9469..eb1300fe137bb0f8e4ee0029b9f90524b09415a3 100644
--- forkSrcPrefix/src/ipc-types.d.ts
+++ forkDstPrefix/src/ipc-types.d.ts
@@ -11,6 +11,8 @@ interface ShowNotificationOptions extends Electron.NotificationConstructorOption
 	showIcon: boolean;
 }
 
+type ImportState = 'initial' | 'importing' | 'imported';
+
 interface StoppedSiteDetails {
 	running: false;
 
@@ -29,7 +31,7 @@ interface StoppedSiteDetails {
 		supportsMenus: boolean;
 	};
 	isAddingSite?: boolean;
-	importState?: 0 | 1 | 2;
+	importState?: ImportState;
 }
 
 interface StartedSiteDetails extends StoppedSiteDetails {
diff --git forkSrcPrefix/src/hooks/use-site-details.tsx forkDstPrefix/src/hooks/use-site-details.tsx
index 3a908384933900fce168840cbcb66d1a77aa5b5a..b56304c087fc5ba9bd6565765a93eca4be2b5a0a 100644
--- forkSrcPrefix/src/hooks/use-site-details.tsx
+++ forkDstPrefix/src/hooks/use-site-details.tsx
@@ -10,7 +10,7 @@ import {
 	useState,
 } from 'react';
 import { getIpcApi } from '../lib/get-ipc-api';
-import { BackupArchiveInfo, ImportState } from '../lib/import-export/import/types';
+import { BackupArchiveInfo } from '../lib/import-export/import/types';
 import { sortSites } from '../lib/sort-sites';
 import { useSnapshots } from './use-snapshots';
 
@@ -227,11 +227,11 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 	);
 
 	const importFile = useCallback( async ( file: BackupArchiveInfo, selectedSite: SiteDetails ) => {
-		let finalImportState = ImportState.Imported;
+		let finalImportState: ImportState = 'imported';
 		try {
 			setData( ( prevSites ) =>
 				prevSites.map( ( site ) =>
-					site.id === selectedSite.id ? { ...site, importState: ImportState.Importing } : site
+					site.id === selectedSite.id ? { ...site, importState: 'importing' } : site
 				)
 			);
 			const backupFile: BackupArchiveInfo = {
@@ -243,7 +243,7 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 				title: selectedSite.name,
 				body: __( 'Import complete' ),
 			} );
-			finalImportState = ImportState.Imported;
+			finalImportState = 'imported';
 		} catch ( error ) {
 			getIpcApi().showMessageBox( {
 				type: 'error',
@@ -253,7 +253,7 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 				),
 				buttons: [ __( 'OK' ) ],
 			} );
-			finalImportState = ImportState.Initial;
+			finalImportState = 'initial';
 		} finally {
 			setData( ( prevSites ) =>
 				prevSites.map( ( site ) =>

```

Reply:
I don't have a strong preference over the two and the current approach, so feel free to pick any option.

Reply:
Thanks for trying different alternatives, I think using string literals is a better approach. I've applied the changes here https://github.com/Automattic/studio/pull/369/commits/eca0bdb6d313e04a328b728af09e263e7ae68289
I've also removed the 'initial' state because that can be identified when the state is different from importing or imported and by default is undefined.